File: frontend/src/components/Carousel/items/ServiceNow.js
/**
 * Author: Deric Le
 * Description: Handles ServiceNow API
 */

const FetchSNOW = async () => {
    try {
        const fetchCalls = await fetch('http://localhost:8000/api/SNOW/calls');
        if (!fetchCalls.ok) { throw new Error('Failed to fetch calls from SNOW'); }

        const calls = await fetchCalls.json();
        return calls;

    } catch (error) { 
        console.error('Error fetching data:', error);
        return null;
    }
    
};

export default FetchSNOW;

File: backend/capops_api/views/servicenow.py
"""
Author: Deric Le
Description: Define functions to be used for ServiceNow API
"""

import random 

from django.shortcuts import render
from rest_framework.decorators import api_view
from rest_framework.response import Response

# accessible via GET request
# see ~/frontend/src/components/HelloWorld.js to see how it's used
@api_view(['GET'])
def getCalls(request):
    calls = random.randint(0, 10)
    return Response({'data': calls})